File: src/ros2_bridge/mappers/sensor_msgs/relative_humidity.rs
//! Typed mapper for `sensor_msgs/msg/RelativeHumidity`.

use crate::ros2_bridge::mapper::TypedTopicMapper;

pub(crate) fn relative_humidity_to_bus(msg: ros_env::sensor_msgs::msg::RelativeHumidity) -> crate::sensor_msgs::msg::v1::RelativeHumidity {
    crate::sensor_msgs::msg::v1::RelativeHumidity {
        header: Some(crate::ros2_bridge::mappers::std_msgs::header::header_to_bus(msg.header)),
        relative_humidity: msg.relative_humidity,
        variance: msg.variance,
    }
}

pub(crate) fn relative_humidity_to_ros(bus: crate::sensor_msgs::msg::v1::RelativeHumidity) -> ros_env::sensor_msgs::msg::RelativeHumidity {
    ros_env::sensor_msgs::msg::RelativeHumidity {
        header: crate::ros2_bridge::mappers::std_msgs::header::header_to_ros(bus.header.unwrap_or_default()),
        relative_humidity: bus.relative_humidity,
        variance: bus.variance,
    }
}

#[derive(Clone, Copy, Debug, Default)]
pub struct SensorMsgsRelativeHumidityMapper;

impl TypedTopicMapper for SensorMsgsRelativeHumidityMapper {
    type Ros = ros_env::sensor_msgs::msg::RelativeHumidity;
    type Bus = crate::sensor_msgs::msg::v1::RelativeHumidity;

    fn ros_to_bus(&self, msg: Self::Ros) -> crate::errors::Result<Self::Bus> {
        Ok(relative_humidity_to_bus(msg))
    }

    fn bus_to_ros(&self, msg: Self::Bus) -> crate::errors::Result<Self::Ros> {
        Ok(relative_humidity_to_ros(msg))
    }
}

File: bindings/python/robot_bus/ros2_bridge/mappers/sensor_msgs/relative_humidity.py
"""Generated mapper for `sensor_msgs/msg/RelativeHumidity`."""

from __future__ import annotations

from robot_bus.ros2_bridge.mappers import _convert
from robot_bus.ros2_bridge.mappers.std_msgs.header import header_to_bus, header_to_ros

def relative_humidity_to_bus(msg):
    from robot_bus.sensor_msgs.msg.v1 import RelativeHumidity as BusMsg

    bus = BusMsg()
    bus.header.CopyFrom(header_to_bus(msg.header))
    bus.relative_humidity = msg.relative_humidity
    bus.variance = msg.variance
    return bus


def relative_humidity_to_ros(bus):
    from sensor_msgs.msg import RelativeHumidity as RosMsg

    out = RosMsg()
    out.header = header_to_ros(bus.header)
    out.relative_humidity = bus.relative_humidity
    out.variance = bus.variance
    return out


class SensorMsgsRelativeHumidityMapper:
    def ros_msg_type(self):
        from sensor_msgs.msg import RelativeHumidity as RosMsg

        return RosMsg

    def ros_to_bus(self, msg) -> bytes:
        return relative_humidity_to_bus(msg).SerializeToString()

    def bus_to_ros(self, payload: bytes):
        from robot_bus.sensor_msgs.msg.v1 import RelativeHumidity as BusMsg

        bus = BusMsg()
        bus.ParseFromString(payload)
        return relative_humidity_to_ros(bus)
